### ISSUE TITLE
The size of tensor a (3) must match the size of tensor b (9) at non-singleton dimension 0

### ISSUE THREAD

**Issue by Peter-D-James** [salesforce/BLIP#165]
![image](https://github.com/salesforce/BLIP/assets/42126680/b32182e6-e3bc-45d1-b87e-94f6f0738867)
When I use beam search to generate caption for a picture on Colab, this error occurs. My transformers version is 4.25.1. And it works when I use nucleus sampling. How should I solve this

**Comment by jding25** [comment]
<img width="793" alt="Screen Shot 2023-07-09 at 9 38 26 AM" src="https://github.com/salesforce/BLIP/assets/92660454/19c6df4b-2cd3-4246-9422-37f52d2b2d86">
I commented out the line that causes error and uncommented the line below as a temporary solution

**Comment by Peter-D-James** [comment]
> <img alt="Screen Shot 2023-07-09 at 9 38 26 AM" width="793" src="https://user-images.githubusercontent.com/92660454/252121434-19c6df4b-2cd3-4246-9422-37f52d2b2d86.png"> I commented out the line that causes error and uncommented the line below as a temporary solution

Yeah, that's what I did. But it's strange that I can use these two methods in the official demo Colab notebook, but I can't use beam search when I write the code myself.

**Comment by Peter-D-James** [comment]
> <img alt="Screen Shot 2023-07-09 at 9 38 26 AM" width="793" src="https://user-images.githubusercontent.com/92660454/252121434-19c6df4b-2cd3-4246-9422-37f52d2b2d86.png"> I commented out the line that causes error and uncommented the line below as a temporary solution

And I succeeded after downgrading the transformers version to 4.16.0. But it seems that I cannot import AutoProcessor when using this version of transformers.

**Comment by HWHHWHNO1** [comment]
same issue

**Comment by csf0429** [comment]
same issue， waiting for fixing

**Comment by Saint-lsy** [comment]
same issue

**Comment by bmaltais** [comment]
I suggest opening an issue with kohya_ss in his repo as he is the one maintaining the behind the scene code. I only wrap his script in a GUI: https://github.com/kohya-ss/sd-scripts

**Comment by shams2023** [comment]
> ![image](https://user-images.githubusercontent.com/42126680/251770843-b32182e6-e3bc-45d1-b87e-94f6f0738867.png) When I use beam search to generate caption for a picture on Colab, this error occurs. My transformers version is 4.25.1. And it works when I use nucleus sampling. How should I solve this

Dude, I also ran into this problem when I changed sample = False to sample = True like this:
![image](https://github.com/salesforce/BLIP/assets/141383792/bde5642b-714f-4e62-88e4-a5aa083ad90e)

This will allow for successful execution. But I wonder why beam search doesn't work and only nuclear search can be used

**Comment by shams2023** [comment]
> > ![image](https://user-images.githubusercontent.com/42126680/251770843-b32182e6-e3bc-45d1-b87e-94f6f0738867.png) When I use beam search to generate caption for a picture on Colab, this error occurs. My transformers version is 4.25.1. And it works when I use nucleus sampling. How should I solve this
> 
> Dude, I also ran into this problem when I changed sample = False to sample = True like this: ![image](https://user-images.githubusercontent.com/141383792/268652932-bde5642b-714f-4e62-88e4-a5aa083ad90e.png)
> 
> This will allow for successful execution. But I wonder why beam search doesn't work and only nuclear search can be used
![image](https://github.com/salesforce/BLIP/assets/141383792/c44532c7-2fdf-4af1-93da-b1c391ac30e7)

**Comment by hannahgym** [comment]
I have same problem and  using transformers=4.25.1  did not help

**Comment by LWShowTime** [comment]
> I found that when commented out the line in /model/blip.py line 131 fix the problem:
> 
> ![image](https://private-user-images.githubusercontent.com/57023743/287519088-bf843a0a-a070-4dde-ab87-db0cb25da79c.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MDQ2NDExMjMsIm5iZiI6MTcwNDY0MDgyMywicGF0aCI6Ii81NzAyMzc0My8yODc1MTkwODgtYmY4NDNhMGEtYTA3MC00ZGRlLWFiODctZGIwY2IyNWRhNzljLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDAxMDclMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQwMTA3VDE1MjAyM1omWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPWViNzI5YmEyMzAyODlkNzkyNTNhMGM3NmNhYjg3M2UzYjVhOWEzZTMyMGM1NGY4ZDAyNTc0MDRjNWM1MTM3YmEmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.7-OQTQIEJPKQPxqBquhAneH9SOPuBwDmrgW_kPQLofk)
> 
> Don't know why, hope someone can provide the detail explanation down the hood.

When you comment out this line, the dimension 9 will be 3, so it can run. But this is not the beam search since you just keep one result!

**Comment by mandalinadagi** [comment]
in order to solve this problem you need to set num_beams=1 not 3. (for instance in blip_vqa.py line 92)

**Comment by LWShowTime** [comment]
I solved this problem, if the transformers is 4.16.0, everything is ok. But I used the transformers4.36.2, in this case, in the 818 lines of the generation_utils.py of transformers:
you need to comment out the _expand_dict_for_generation function where the encoder_hidden_state was multiplied again with the beam search num! 
Finally solved!!!

**Comment by bmaltais** [comment]
You should submit a PR to Koby’s in his sd-scripts repo to fix it for good.

**Comment by amztc34283** [comment]
Updating to 1.0.2 fixed it for me.

**Comment by Ashh-Z** [comment]
I used Transformers==4.17 and did not face further issues

**Comment by 792864625** [comment]
> > > ![image](https://user-images.githubusercontent.com/42126680/251770843-b32182e6-e3bc-45d1-b87e-94f6f0738867.png) When I use beam search to generate caption for a picture on Colab, this error occurs. My transformers version is 4.25.1. And it works when I use nucleus sampling. How should I solve this
> > 
> > 
> > Dude, I also ran into this problem when I changed sample = False to sample = True like this: ![image](https://user-images.githubusercontent.com/141383792/268652932-bde5642b-714f-4e62-88e4-a5aa083ad90e.png)
> > This will allow for successful execution. But I wonder why beam search doesn't work and only nuclear search can be used
> > ![image](https://github.com/salesforce/BLIP/assets/141383792/c44532c7-2fdf-4af1-93da-b1c391ac30e7)

Thank U Bro

**Comment by chilljudaoren** [comment]
> I used Transformers==4.17 and did not face further issues

the ans is right!
